client/src/components/profileInfoEdit/ProfileInfoEdit.js

```diff
@@ -64,14 +68,91 @@ const ProfileInfoEdit = () => {
     }
   };
 
+  const setPassword = (field, value) => {
+    setprofileDataPassword({
+      ...profileDataPassword,
+      [field]: value,
+    });
+    if (errors[field]) {
+      setErrors({
+        ...errors,
+        [field]: null,
+      });
+    }
+  };
+
+  const passwordSchema = yup.object().shape({
+    password: yup
+      .string()
+      .required('Please Enter your password')
+      .matches(
+        /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/,
+        'Use 8 or more characters with a mix of letters, numbers & symbols',
+      ),
+    confirmedPassword: yup
+      .string(),
+  });
+
+  const checkPassword = async () => {
+    const { password, confirmedPassword } = profileDataPassword;
+
+    const valid = await passwordSchema.isValid({
```

Comment:
It will automatically checks for you if you pass the restriction inside the schema.

Reply:
this is needed as im not using formik